Rec cal cost test

JasonPaulGithub opened the pull request:
## Description

reviewpad:summary

oetyng commented on sn_node/src/node/flow_ctrl/dispatcher.rs:
```diff
@@ -104,19 +109,33 @@ impl Dispatcher {
                 traceroute,
-            } => {
-                let mut node = self.node.write().await;
+            } => match msg {
```
It's `self.dysfunction_tracking.track_issue(` in `fn read_data_from_adults` that requires the write lock.
But now there is a new `TrackNodeIssueInDysfunction` for it, so no need to write lock on a read, and no need for the handle_valid_query_msg` fn, just issue that cmd and it will be dispatched to the write locked node.

Replace line 131 in node/data/records/mod.rs with

cmds.push(Cmd::TrackNodeIssueInDysfunction {
                                name: target.name(),
                                issue: IssueType::PendingRequestOperation(Some(operation_id)),
                            });

davidrusu commented on sn_interface/src/types/cache/mod.rs:
```diff
@@ -138,18 +133,18 @@ where
             let excess_keys: Vec<_>;
             {
-                let read_items = self.items.read().await;
+                let read_items = &self.items;
                 let mut items = read_items.iter().collect_vec();
```
No need for a let binding for the reference anymore.
```suggestion
                let mut items = self.items.iter().collect_vec();
```

davidrusu commented on sn_node/examples/routing_stress.rs:
```diff
@@ -752,55 +752,51 @@ struct ProbeTracker {
             .sections
-            .iter()
-            .find_map(|(prefix, cache)| block(cache.get(dst)).map(|pk| (prefix, pk, cache)));
+            .iter_mut()
```
Does this need to be `iter_mut()`

joshuef commented:
bors r+

RolandSherwin committed:
feat(manager): provide option to start the manager as a daemon

bochaco committed:
feat(folders): keep track of local changes to Folders

RolandSherwin committed:
fix(manager): fix sync issue while trying to use trait objects

RolandSherwin committed:
fix(manager): don't specify user while spawning daemon

RolandSherwin committed:
feat(manager): add rpc call to restart node service and process

bochaco committed:
feat(folders): realise local changes made to folders/files

iancoleman committed:
feat(faucet): claim using signature of safe wallet

The bitcoin maid address is used to sign a safe wallet address.
Distributions are sent to the signed safe wallet address.
This removes the need for ecies encryption/decryption.
This also removes the need to have the maid private key available.

RolandSherwin committed:
refactor(manager): removing support for process restarts

RolandSherwin committed:
feat(manager): stop the daemon if it is already running

RolandSherwin committed:
feat(protocol): add daemon socket addr to node registry

jacderida committed:
feat: provide `faucet add` command

A new `faucet` subcommand is introduced, enabling the faucet to be controlled as a service. This is
necessary because as with nodes, the faucet also needs to be upgraded on a new deployment; it may
also require downgrading in the event of a problem.

The existing `faucet` command, which was created as a utility for the backwards compatibility test,
is removed; now that we can define the faucet as a service, it will no longer be necessary.

The function that determines the data directory for the faucet was being duplicated in two different
crates, and the node manager would have added a third, so it was refactored as public in
`sn_transfers` and shared from there.

jacderida committed:
feat: provide `faucet start` command

This command is quite similar to the `start` command for starting a node service, but my judgement
is there probably isn't enough code that is common between them to extract out something more
generic.

BREAKING CHANGE: the `faucet_pid` field of the node registry was removed in favour of a `faucet`
field, which provides more details, including the PID.

jacderida committed:
refactor: create a `faucet_control` module

Rather than having a `faucet` module within `node_control`, I thought it made sense to also have a
`faucet_control` module, and I've organised it the same as `node_control`.

RolandSherwin committed:
fix(deps): don't add unix dep to whole crate

RolandSherwin committed:
feat(test): imporve restart api for tests

RolandSherwin committed:
feat(test): add option to retain_peer_id for the node's restart rpc cmd

RolandSherwin committed:
feat(daemon): retain peer_id while restarting a safenode service

RolandSherwin committed:
fix(daemon): set the proper safenode path while restarting a service

RolandSherwin committed:
fix(daemon): create node service dir while restarting as new peer

RolandSherwin committed:
chore(daemon): add more context to errors

RolandSherwin committed:
feat(release): package and release daemon binary and store it to safenode-manager bucket

RolandSherwin committed:
test(manager): add daemon restart test

bochaco committed:
feat(folders): regenerate tracking info when downloading Folders fm the network

jacderida committed:
feat: provide `faucet stop` command

As with the `faucet start` command, this provides the corresponding `stop` command. I revisited the
idea of sharing code between the node and faucet services, but again came to the conclusion that
there wasn't much value added for the complexity it introduced. The result is that we have code
between them that only has small differences, but I think this is OK for now.

bochaco committed:
chore(folders): minor improvements to folders status report

maqi committed:
feat(node): record_store chunk in batch and setup distance_range

grumbach committed:
feat: expose sn related deps to app builders

grumbach committed:
feat: collect royalties through DAG

grumbach committed:
feat: batch royalties redemption

JasonPaulGithub committed:
feat: file to download

bochaco committed:
feat(folders): store files data-map within Folders metadata chunk

bochaco committed:
feat(folders): avoid chunking files when retrieving them with Folders from the network

bochaco committed:
refactor(folders): some simplifications to acc-packet codebase

maqi committed:
fix: record_store no longer update distance_range via close_group change

grumbach committed:
fix: double spend case identified by Shu

RolandSherwin committed:
chore: revert wrong rebase changes

RolandSherwin committed:
refactor(daemon): rename daemon binary to safenodemand

RolandSherwin committed:
feat(manager): add subcommands for daemon

JasonPaulGithub committed:
feat: removal of some messages from vody body

RolandSherwin committed:
fix(test): provide absolute path for daemon restart test

JasonPaulGithub committed:
feat: split upload and upload with iter

JasonPaulGithub committed:
feat: a more clear param for a message function

JasonPaulGithub committed:
feat: gitignore to alpha

grumbach committed:
fix: filter out spent cashnotes in received client transfers

grumbach committed:
feat: actionable double spend reporting

grumbach committed:
ci: fix royalties gossip test

joshuef committed:
chore: fix node-man deps for transfers

maqi committed:
feat(node): bad verification to exclude connections from bad_nodes

JasonPaulGithub committed:
feat: refactor upload with iter

grumbach committed:
feat: genesis double spend test

maqi committed:
feat(node): exponential pricing when storage reaches high

JasonPaulGithub committed:
docs: get clippy to stop mentioning this

iancoleman committed:
feat(cli): files download respects filename path

Instead of downloading to dirs_next::download_dir() the path specified
by the input is respected, either relative or absolute path.

eg
`safe files download myfile.txt ADDR`
Destination before this PR: $HOME/download/myfile.txt
Destination after this PR: $PWD/myfile.txt

`safe files download my/relative/dir/myfile.txt ADDR`
Destination before this PR: $HOME/download/my/relative/dir/myfile.txt
Destination after this PR: $PWD/my/relative/dir/myfile.txt

`safe files download /tmp/myfile.txt ADDR`
Destination before this PR: $HOME/download/tmp/myfile.txt
Destination after this PR: /tmp/myfile.txt

Discussion of this by community:
https://safenetforum.org/t/newyearnewnet-04-01-2024-testnet-offline/39034/455

JasonPaulGithub committed:
chore: rename to iterative upload

JasonPaulGithub committed:
chore: rename to iterative upload

joshuef committed:
chore(release): sn_transfers-v0.16.1

JasonPaulGithub committed:
Merge branch 'alpha' of https://github.com/maidsafe/safe_network into alpha

# Conflicts:
#	sn_cli/src/subcommands/acc_packet.rs
#	sn_cli/src/subcommands/files/iterative_uploader.rs
#	sn_cli/src/subcommands/files/upload.rs
#	sn_cli/src/subcommands/folders.rs
#	sn_client/src/folders.rs
#	sn_node_manager/Cargo.toml

JasonPaulGithub committed:
Merge branch 'alpha' of https://github.com/maidsafe/safe_network into alpha

# Conflicts:
#	Cargo.lock
#	sn_cli/Cargo.toml
#	sn_client/Cargo.toml
#	sn_faucet/Cargo.toml
#	sn_networking/Cargo.toml
#	sn_node/Cargo.toml
#	sn_node_manager/Cargo.toml
#	sn_node_rpc_client/Cargo.toml
#	sn_protocol/Cargo.toml
#	sn_transfers/CHANGELOG.md
#	sn_transfers/Cargo.toml

JasonPaulGithub committed:
Merge branch 'maidsafe:alpha' into alpha

JasonPaulGithub committed:
chore: a unit test for the records calculation cost

JasonPaulGithub committed:
Merge branch 'alpha' into rec_cal_cost_test

JasonPaulGithub committed:
chore: cargo fmt

maqi reviewed (changes requested):
commit history (62 commits) is bit confusing.

JasonPaulGithub committed:
chore: values changed for test
